lib/components/src/blocks/ArgsTable/ArgControl.tsx

```diff
@@ -18,22 +18,39 @@ export interface ArgControlProps {
   updateArgs: (args: Args) => void;
 }
 
+const Controls: Record<string, FC> = {
+  array: ObjectControl,
+  object: ObjectControl,
+  boolean: BooleanControl,
+  color: ColorControl,
+  date: DateControl,
+  number: NumberControl,
+  check: OptionsControl,
+  'inline-check': OptionsControl,
+  radio: OptionsControl,
+  'inline-radio': OptionsControl,
+  select: OptionsControl,
+  'multi-select': OptionsControl,
+  range: RangeControl,
+  text: TextControl,
+  file: FilesControl,
+};
+
 const NoControl = () => <>-</>;
 
 export const ArgControl: FC<ArgControlProps> = ({ row, arg, updateArgs }) => {
   const { key, control } = row;
 
   const [isFocused, setFocused] = useState(false);
-  // box because arg can be a fn (e.g. actions) and useState calls fn's
-  const [boxedValue, setBoxedValue] = useState({ value: arg });
+  const [value, setValue] = useState(() => arg);
```

Comment:
what's the point of doing this:

```ts
useState(() => arg);
```

Reply:
I "boxed" the arg because if the arg was a function it was getting executed, but the boxed version fixes that. However, the function is better than boxing because it doesn't allocate extra memory

Reply:
Please re-add a comment explaining why we're doing "lazy" state initialization here.

Reply:
```suggestion
  // this is because the value can be a function
  const [value, setValue] = useState(() => arg);
```